core/src/avm2/globals/vector.rs

```diff
@@ -284,21 +288,17 @@ pub fn concat<'gc>(
             continue;
         };
 
-        for val in old_vec {
-            if let Ok(val_obj) = val.coerce_to_object(activation) {
-                if !val.is_of_type(activation, val_class) {
-                    let other_val_class = val_obj.instance_class();
-                    return Err(format!(
-                        "TypeError: Cannot coerce Vector value of type {:?} to type {:?}",
```

Comment:
oh, it's not a real error in FP?

Reply:
The `coerce_to_type` should throw the error anyway; `Vector/typechecking` tests this